force-app/main/default/classes/SOQL.cls

```diff
@@ -345,17 +432,38 @@ public with sharing class SOQL {
      */
     public static void setPrimaryContact() {
         //Use the getOpportunityDetailsForClosedWon() method to get the Opportunities
+        List<Opportunity> oppList = getOpportunityDetailsForClosedWon();
+        System.debug(oppList);
 
         // Create a set of Account Ids from the Opportunities
+        Set<Id> accountIdSet = new Set<Id>();
+            for(Opportunity eachOpp : oppList){
+                accountIdSet.add(eachOpp.AccountId);
+            }
 
         // Query the Contacts
+        List<Contact> contactList = [SELECT Id, Name,AccountId FROM Contact WHERE AccountId IN :accountIdSet ORDER BY Name ASC LIMIT 1 ];
+        System.debug(contactList);
 
         // Loop through each Contact and add the Account Id and Contact to the map
+        Map<Id,Id> contactAccountMap = new Map<Id,Id>();
+        for(Contact eachContact : contactList){
+            System.debug(eachContact);
+            contactAccountMap.put(eachContact.AccountId, eachContact.Id);
+            System.debug(contactAccountMap);
+        }
```

Comment:
Also very good. Is there away to query for both accounts and contacts at the same time?